File: csharp/resistor-color/ResistorColor.cs
using System;
using System.Linq;

public static class ResistorColor
{
    public static string[] colors = { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" };
    public static string[] Colors() => colors;
    public static int ColorCode(string color) => Array.IndexOf(Colors(), color);
}

File: java/resistor-color/src/main/java/ResistorColor.java
class ResistorColor {

    enum ColorList {
        black(0),
        brown(1),
        red(2),
        orange (3),
        yellow (4),
        green (5),
        blue(6),
        violet (7),
        grey (8),
        white (9);

        final int resistorCode;
        ColorList(int resistorCode) {
            this.resistorCode = resistorCode;
        }
    }
    int colorCode(String color) {
        return ColorList.valueOf(color).resistorCode;
    }

    String[] colors() {
        String[] colorArray = new String[ColorList.values().length];
        for (ColorList color : ColorList.values()) {
            colorArray[ColorList.valueOf(color.name()).resistorCode] = color.name();
        }
        return colorArray;
    }
}
